### PR TITLE
feat: Upload & Revoke Certificate

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This PR introduces the final 2 buttons in the row, uploading and revoking certificates. The upload button shows a modal that's almost identical to the submit CSR form. The revoke button shows a confirmation modal. I've bundled these together into this PR because they are related features and not too large since their underlying logic has already been written.

Depends on #32 

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ui/src/app/certificate_requests/components.tsx
Comment (gruyaume) [inline ui/src/app/certificate_requests/components.tsx:1]: If i don't remove the newline I'm getting an "Invalid Certificate" Notification

File: ui/src/app/certificate_requests/components.tsx [hunk 1]
```diff
@@ -28,6 +31,88 @@ export function ConfirmationModal({ modalData, setModalData }: ConfirmationModal
...
+                </header>
+                <form className="p-form p-form--stacked">
+                    <div className="p-form__group row">
+                        <label htmlFor="textarea">
+                            Enter or upload the Certificate in PEM format below
+                        </label>
+                        <textarea id="csr-textarea" name="textarea" rows={10} placeholder="-----BEGIN CERTIFICATE-----" onChange={handleTextChange} value={certificatePEMString} />
+                    </div>
+                    <div className="p-form__group row">
+                        <input type="file" name="upload" accept=".pem" onChange={handleFileChange}></input>
```
Comment (gruyaume) [inline ui/src/app/certificate_requests/components.tsx:102]: It is quite common to see `.crt` too

File: ui/src/app/certificate_requests/components.tsx
Comment (gruyaume) [inline ui/src/app/certificate_requests/components.tsx:1]: If we Upload a certificate that was already uploaded we should likely grey out the submit button and mention to the user that "This certificate was already uploaded for this request" or something along those lines

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: ui/src/app/certificate_requests/components.tsx (hunk 1 above)
Comment (kayra1) [reply ui/src/app/certificate_requests/components.tsx:102]: added

File: ui/src/app/certificate_requests/components.tsx
Comment (kayra1) [reply ui/src/app/certificate_requests/components.tsx:1]: should be fixed now

Comment (kayra1) [reply ui/src/app/certificate_requests/components.tsx:1]: added a custom message

Comment (gruyaume) [reply ui/src/app/certificate_requests/components.tsx:1]: I still see the button as being enabled
image

Comment (gruyaume) [reply ui/src/app/certificate_requests/components.tsx:1]: Actually it depends on whether or not I add newlines, we should probably strip all of those

Comment (kayra1) [reply ui/src/app/certificate_requests/components.tsx:1]: ah, forgot one more .trim() call :D

File: ui/src/app/certificate_requests/components.tsx [hunk 2]
```diff
@@ -28,6 +31,95 @@ export function ConfirmationModal({ modalData, setModalData }: ConfirmationModal
...
 
+function SubmitCertificate({ existingCSRText, existingCertText, certText, onClickFunc }: { existingCSRText: string, existingCertText: string, certText: string, onClickFunc: any }) {
+    let certIsValid = false
+    try {
+        extractCert(certText.trim())
+        if (csrMatchesCertificate(existingCSRText.trim(), certText.trim())) {
+            certIsValid = true
+        }
+    }
+    catch { }
```
Comment (gruyaume) [inline ui/src/app/certificate_requests/components.tsx:42]: Here we see "Invalid Certificates" when the certificate is valid but does not match the CSR. I believe we should differentiate those 2 situations with independent error messages.
- The Certificate is invalid
- The Certificate does not match the request

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: ui/src/app/certificate_requests/components.tsx (hunk 2 above)
Comment (kayra1) [reply ui/src/app/certificate_requests/components.tsx]: done

**Review by gruyaume — APPROVED** [review]

**Pull request merged**